buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -52,9 +53,12 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Returns a closure of common configuration shared by unit and integration tests. */
     static Closure commonTestConfig(Project project) {
+        def javaHome = System.getProperty("java.home")
+        if (!javaHome) {
+            throw new InvalidUserDataException("Couldn't locate 'java.home'")
```

Comment:
[`java.home` is a required property](http://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getProperties()). It will always be set.